packages/Radio/src/components/Group/Group.js

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import PropTypes from "prop-types";
+import nanoid from "nanoid";
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import RadioContext from "../../RadioContext";
+
+const propTypes = {
+  /** aria-labelledby prop on the containing group element */
+  a11yText: PropTypes.string,
+
+  /** Can deselect any radio */
+  canDeselect: PropTypes.bool,
+
+  /** The individual radio items. */
+  children: PropTypes.node,
+
+  /** Are all radios disabled */
+  isDisabled: PropTypes.bool,
+
+  /** On change of radio selection. */
+  onChange: PropTypes.func.isRequired,
+
+  /** The size for all radio components. */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+};
+
+const defaultProps = {
+  a11yText: "",
+  canDeselect: false,
+  children: null,
+  isDisabled: false,
+  size: ShirtSizes.MEDIUM,
+};
+
+function Group(props) {
+  const { a11yText, canDeselect, children, isDisabled, onChange, ...moreGroupProps } = props;
+  const defaultCheckedIndex = React.Children.toArray(children).findIndex(child => child.props.defaultIsChecked);
```

Comment:
This still assumes `Radio` is a direct child of `Group` which defeats the purpose of using context. (Recommend to create a story with code like
```js
<Radio.Group>
  <>
    <Radio />
    <Radio />
  <>
</Radio.Group>
```
to see that it doesn't work)
If it's acceptable to put the checked status on the group itself (i.e. `checkedIndex` prop), then I think context can still be used. Otherwise we should revert that change since it's not achieving what we want.

Reply:
I would have thought it is ok to expect the usage of this component is to always use Radio components as the siblings? I understand that there will be issues if they use a different child but is that a scenario that we need to cover? We do provide storybook examples and a readme to show correct usage...

Reply:
We could throw an error if the wrong child type was passed as well..

Reply:
What if you need side-by-side radios though? You'll need some flexbox divs in there for layout.

Reply:
Perhaps we will need to revisit the required api in a future PR. I know in the case of formElement we pass an `isInline` prop to dictate the layout rather than style the children directly,  - https://github.com/acl-services/paprika/blob/master/packages/FormElement/src/FormElement.js#L34 But perhaps we are not too keen on that approach? Will need discussion. For now I have changed it back to use `cloneElement` rather than context. Discussing with @alexzherdev we have established that if we wanted to retain the index to dictate which radio is selected (we are also need to track which radio is deselectable) we would perhaps have to go down the route of https://gist.github.com/ryanflorence/10e9387f633f9d2e6f444a9bddaabf6e#option-4-the-unholy-option.